sequential/seq2pat.py

```diff
@@ -283,12 +294,36 @@ class Seq2Pat:
         sequences but no constraints are used to run the mining efficiently and practically.
         Power users can choose to drop this constraint by setting it to be None or increase the maximum span
         as the system has resources to support.
+    batch_size: Optional[int]
+        When batch_size is set, program runs Seq2Pat on batches of sequences instead of on the entire set of
+        sequences for improving scalability. Each batch contains `batch_size` sequences as a random sample of
+        entire set. A mining task will run on each batch with a reduced minimum row count (min_frequency) threshold.
+        Please refer to description of discount_factor parameter for how min_frequency is reduced.
+        Resulted patterns will be aggregated from the mining results of each batch by calculating the sum of
+        the occurrences. Finally the original minimum row count threshold is applied to the patterns after aggregation.
```

Comment:
for batch_size parameter, add a comment about what's the default behaviour.

Reply:
BTW, I would immediately copy the Experimental Results Summary here.. Taht would greatly help a power user reading the docs. 

* Some experimental analysis for batch_size vs. discount_factor vs. runtime tested on a data set with 100k sequences. 
* The results show that when batch_size increases , e.g. from 10000 to 100000, we observe an increase in runtime, while the mined patterns are all the same as mining on the entire set using single thread. When batch_size=10000, we get the most runtime benefit when it is compared to running on entire set.
* On the same 100k sequences, we set batch_size=10000 and change discount_factor from 0.1 to 1.0. We observe that the runtime decreases as discount_factor increases. Only when discount_factor=1.0, the batching mode will miss some patterns compared to running on entire set. We would recommend discount_factor=0.2 by default for the robustness in results, at the expenses of runtime.
* In an even larger test on ~1M sequences, we set batch_size=10000, discount_factor=0.8, n_jobs=8. Batch mode saves 60% of the runtime compared to running on entire set, while the resulted patterns from the two processes are the same.
* When data size is small, e.g., a few thousand sequences, there is no benefit to run batch mode. Thus, we would recommend using the batch mode only when data has at least hundreds of thousands of sequences for gaining the runtime benefit.